packages/x-tree-view/src/internals/plugins/useTreeViewFocus/useTreeViewFocus.ts

```diff
@@ -89,10 +79,10 @@ export const useTreeViewFocus: TreeViewPlugin<UseTreeViewFocusSignature> = ({
   const createHandleBlur =
     (otherHandlers: EventHandlers) => (event: React.FocusEvent<HTMLUListElement>) => {
       otherHandlers.onBlur?.(event);
-      setFocusedNodeId(null);
+      models.focusedNode.setControlledValue(null);
```

Comment:
I would be in favor of having one single function that calls `setControlledValue`, that way it can also call `onFocusedNodeChange` without any fear of missing on occurence.

Here, I think it should call `onFocusedNodeChange` since it's now the callback of a controllable model.
This would require to change the signature of `onFocusedNodeChange` to accept `null`

Right now in your doc example, when we blur the Tree View, the previously focused element remains visually focused, which is not the intended behavior IMHO.

But this raises another question: if the Tree View is not focused and `focusedNode != null`, should we display the item as focused? I don't think so, I think that `isNodeFocused` should check if the Tree View is focused.
I would like to have the opinion of others on this one @LukasTy @alexfauquette

Reply:
I may lack of context, but from my past experience on the data grid and pickers, each time we tracked a `focused` item, if the component don't have focus, it's not visible, but used to initial the focus when it comes back.

Here is a data grid example of the focus coming back to the last focused cell

https://github.com/mui/mui-x/assets/45398769/cdd810f3-4d27-410d-9a57-902f5e65f774